implementations/rust/ockam/ockam_app/src/app/app_state.rs

```diff
@@ -227,12 +225,38 @@ fn create_model_state_repository(state: CliState) -> Arc<dyn ModelStateRepositor
 }
 
 /// Load a previously persisted ModelState
-fn load_model_state(model_state_repository: Arc<dyn ModelStateRepository>) -> ModelState {
-    match block_on(model_state_repository.load()) {
-        Ok(model_state) => model_state.unwrap_or(ModelState::default()),
-        Err(e) => {
-            println!("cannot load the model state: {e:?}");
-            panic!("{}", e)
+fn load_model_state(
+    model_state_repository: Arc<dyn ModelStateRepository>,
+    node_manager: &mut NodeManager,
+    context: Arc<Context>,
+) -> ModelState {
+    block_on(async {
+        match model_state_repository.load().await {
+            Ok(model_state) => {
+                let model_state = model_state.unwrap_or(ModelState::default());
+                for tcp_outlet in model_state.get_tcp_outlets() {
```

Comment:
I suggest that we start moving this code to a separate function to show that this is a separate concern: updating the `NodeManager` with some configuration data. This will make it easier to extend, and also easier to unit test.

Reply:
100% agree. What do you think about moving it to a `tcp/outlet/model_state` module to avoid cluttering the main `app/model_state` module?